freud/density.pyx

```diff
@@ -528,19 +515,29 @@ cdef class GaussianDensity(Compute):
             The image grid with the Gaussian density.
     """  # noqa: E501
     cdef freud._density.GaussianDensity * thisptr
-    cdef arglist
-
-    def __cinit__(self, *args):
-        if len(args) == 3:
-            self.thisptr = new freud._density.GaussianDensity(
-                args[0], args[1], args[2])
-            self.arglist = [args[0], args[1], args[2]]
-        elif len(args) == 5:
-            self.thisptr = new freud._density.GaussianDensity(
-                args[0], args[1], args[2], args[3], args[4])
-            self.arglist = [args[0], args[1], args[2], args[3], args[4]]
+    cdef float r_max
+
+    def __cinit__(self, width, r_max, sigma):
+        cdef vec3[uint] width_vector
+        if isinstance(width, int):
+            width_vector = vec3[uint](width, width, width)
+        elif isinstance(width, (list, tuple)):
+            if len(width) == 2:
+                width_vector = vec3[uint](width[0], width[1], 1)
+            elif len(width) == 3:
+                width_vector = vec3[uint](width[0], width[1], width[2])
+            else:
+                raise ValueError("The width must be either a number of pixels "
+                                 "or a list of length 3 (2) indicating the "
```

Comment:
I would make it more obvious that this number matches the system dimensionality.